[beta] backport rollup

Mark-Simulacrum opened the pull request:
* save-analysis: Account for async desugaring in async fn return types #65936 
* resolve: Turn the "non-empty glob must import something" error into a lint #65539 
* Updated RELEASES.md for 1.39.0 #64878 (squashed)

XAMPPRocky committed:
Updated RELEASES.md for 1.39.0

traxys committed:
changing non-empty glob must import something to a lint

Xanewok committed:
save-analysis: Account for async desugaring in async fn return types

rust-highfive commented:
:warning: **Warning** :warning:

* Pull requests are usually filed against the master branch for this repo, but this one is against beta. Please double check that you specified the right target!

Mark-Simulacrum commented:
Related PRs have been approved and no merge conflicts were detected during merging, but I'm not prioritizing this or approving myself since it seems like we might have a few backports incoming tomorrow and maybe early this weekend.

pietroalbini commented:
r=me p=10

Mark-Simulacrum commented:
@bors r=pietroalbini p=10

The PR I was thinking about was https://github.com/rust-lang/rust/pull/66018 but it seems like we don't actually need to backport it so I'm going ahead here.

bors commented:
:pushpin: Commit 559ebafa24e1c696a9b35663a2e75db0aadca715 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 559ebafa24e1c696a9b35663a2e75db0aadca715 -->
<!-- homu: {"type":"Approved","sha":"559ebafa24e1c696a9b35663a2e75db0aadca715","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 559ebafa24e1c696a9b35663a2e75db0aadca715 with merge 7077053898a3e148b8c59e29a3b43b6eb66127b6...
<!-- homu: {"type":"BuildStarted","head_sha":"559ebafa24e1c696a9b35663a2e75db0aadca715","merge_sha":"7077053898a3e148b8c59e29a3b43b6eb66127b6"} -->

rust-highfive commented:
Your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=12510&view=logs&jobId=0e47216c-a892-589a-a5b3-5fa9021bc5b1) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/12510/logs/821), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/12510/logs/821)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-11-01T23:42:12.8661757Z  ---> 360c029e77d3
2019-11-01T23:42:12.8662756Z Step 4/18 : RUN bash /scripts/emscripten.sh
2019-11-01T23:42:13.0074841Z  ---> Running in 7a20c5d8dc91
2019-11-01T23:42:13.4764759Z + cd /
2019-11-01T23:42:13.4765272Z + curl -fL https://mozilla-games.s3.amazonaws.com/emscripten/releases/emsdk-portable.tar.gz
2019-11-01T23:42:13.4765499Z + tar -xz
2019-11-01T23:42:13.4799300Z                                  Dload  Upload   Total   Spent    Left  Speed
2019-11-01T23:42:13.4799369Z 
2019-11-01T23:42:13.9464770Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:13.9465997Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:13.9465997Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:13.9537220Z curl: (22) The requested URL returned error: 403 Forbidden
2019-11-01T23:42:13.9555185Z 
2019-11-01T23:42:13.9555872Z gzip: stdin: unexpected end of file
2019-11-01T23:42:13.9556464Z tar: Child returned status 1
2019-11-01T23:42:13.9556794Z tar: Error is not recoverable: exiting now
2019-11-01T23:42:14.3250538Z The command '/bin/sh -c bash /scripts/emscripten.sh' returned a non-zero code: 2
2019-11-01T23:42:15.4348609Z Sending build context to Docker daemon  529.4kB
2019-11-01T23:42:15.4349329Z 
2019-11-01T23:42:15.4556795Z Step 1/18 : FROM ubuntu:16.04
2019-11-01T23:42:15.4563547Z  ---> 5f2bf26e3524
---
2019-11-01T23:42:15.4576481Z  ---> 360c029e77d3
2019-11-01T23:42:15.4576861Z Step 4/18 : RUN bash /scripts/emscripten.sh
2019-11-01T23:42:15.6147324Z  ---> Running in 573e356929ac
2019-11-01T23:42:16.0956883Z + cd /
2019-11-01T23:42:16.0973436Z + curl -fL https://mozilla-games.s3.amazonaws.com/emscripten/releases/emsdk-portable.tar.gz
2019-11-01T23:42:16.0974112Z + tar -xz
2019-11-01T23:42:16.0975331Z                                  Dload  Upload   Total   Spent    Left  Speed
2019-11-01T23:42:16.0975487Z 
2019-11-01T23:42:16.5077996Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:16.5078726Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:16.5078726Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:16.5156592Z curl: (22) The requested URL returned error: 403 Forbidden
2019-11-01T23:42:16.5175011Z 
2019-11-01T23:42:16.5175496Z gzip: stdin: unexpected end of file
2019-11-01T23:42:16.5181913Z tar: Child returned status 1
2019-11-01T23:42:16.5182430Z tar: Error is not recoverable: exiting now
2019-11-01T23:42:16.8433726Z The command '/bin/sh -c bash /scripts/emscripten.sh' returned a non-zero code: 2
2019-11-01T23:42:18.9435770Z Sending build context to Docker daemon  529.4kB
2019-11-01T23:42:18.9436567Z 
2019-11-01T23:42:18.9720137Z Step 1/18 : FROM ubuntu:16.04
2019-11-01T23:42:18.9721733Z  ---> 5f2bf26e3524
---
2019-11-01T23:42:18.9733632Z  ---> 360c029e77d3
2019-11-01T23:42:18.9734010Z Step 4/18 : RUN bash /scripts/emscripten.sh
2019-11-01T23:42:19.1735501Z  ---> Running in 238013d16f8d
2019-11-01T23:42:19.6043880Z + cd /
2019-11-01T23:42:19.6061571Z + tar -xz
2019-11-01T23:42:19.6061963Z + curl -fL https://mozilla-games.s3.amazonaws.com/emscripten/releases/emsdk-portable.tar.gz
2019-11-01T23:42:19.6120945Z                                  Dload  Upload   Total   Spent    Left  Speed
2019-11-01T23:42:19.6121284Z 
2019-11-01T23:42:20.0325373Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:20.0326559Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:20.0326559Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:20.0397078Z curl: (22) The requested URL returned error: 403 Forbidden
2019-11-01T23:42:20.0412009Z 
2019-11-01T23:42:20.0412291Z gzip: stdin: unexpected end of file
2019-11-01T23:42:20.0421106Z tar: Child returned status 1
2019-11-01T23:42:20.0421405Z tar: Error is not recoverable: exiting now
2019-11-01T23:42:20.4354114Z The command '/bin/sh -c bash /scripts/emscripten.sh' returned a non-zero code: 2
2019-11-01T23:42:23.5467078Z Sending build context to Docker daemon  529.4kB
2019-11-01T23:42:23.5467331Z 
2019-11-01T23:42:23.5678510Z Step 1/18 : FROM ubuntu:16.04
2019-11-01T23:42:23.5680501Z  ---> 5f2bf26e3524
---
2019-11-01T23:42:23.5692563Z  ---> 360c029e77d3
2019-11-01T23:42:23.5694558Z Step 4/18 : RUN bash /scripts/emscripten.sh
2019-11-01T23:42:23.7264079Z  ---> Running in 1386ba20e87b
2019-11-01T23:42:24.1576282Z + cd /
2019-11-01T23:42:24.1582281Z + curl -fL https://mozilla-games.s3.amazonaws.com/emscripten/releases/emsdk-portable.tar.gz
2019-11-01T23:42:24.1582523Z + tar -xz
2019-11-01T23:42:24.1642088Z                                  Dload  Upload   Total   Spent    Left  Speed
2019-11-01T23:42:24.1642149Z 
2019-11-01T23:42:24.5702957Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:24.5703474Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:24.5703474Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:24.5776308Z curl: (22) The requested URL returned error: 403 Forbidden
2019-11-01T23:42:24.5792778Z 
2019-11-01T23:42:24.5792886Z gzip: stdin: unexpected end of file
2019-11-01T23:42:24.5793283Z tar: Child returned status 1
2019-11-01T23:42:24.5793368Z tar: Error is not recoverable: exiting now
2019-11-01T23:42:24.9364289Z The command '/bin/sh -c bash /scripts/emscripten.sh' returned a non-zero code: 2
2019-11-01T23:42:29.0626450Z Sending build context to Docker daemon  529.4kB
2019-11-01T23:42:29.0626809Z 
2019-11-01T23:42:29.0877770Z Step 1/18 : FROM ubuntu:16.04
2019-11-01T23:42:29.0887238Z  ---> 5f2bf26e3524
---
2019-11-01T23:42:29.0892551Z  ---> 360c029e77d3
2019-11-01T23:42:29.0892630Z Step 4/18 : RUN bash /scripts/emscripten.sh
2019-11-01T23:42:29.2781668Z  ---> Running in 01c937095830
2019-11-01T23:42:29.7640957Z + cd /
2019-11-01T23:42:29.7649361Z + curl -fL https://mozilla-games.s3.amazonaws.com/emscripten/releases/emsdk-portable.tar.gz
2019-11-01T23:42:29.7649689Z + tar -xz
2019-11-01T23:42:29.7675842Z                                  Dload  Upload   Total   Spent    Left  Speed
2019-11-01T23:42:29.7675919Z 
2019-11-01T23:42:30.1250567Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:30.1996729Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:30.1996729Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:30.1997160Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2019-11-01T23:42:30.2079904Z curl: (22) The requested URL returned error: 403 Forbidden
2019-11-01T23:42:30.2095914Z 
2019-11-01T23:42:30.2096038Z gzip: stdin: unexpected end of file
2019-11-01T23:42:30.2096281Z tar: Child returned status 1
2019-11-01T23:42:30.2096368Z tar: Error is not recoverable: exiting now
2019-11-01T23:42:30.5841472Z The command '/bin/sh -c bash /scripts/emscripten.sh' returned a non-zero code: 2
2019-11-01T23:42:30.5889174Z 
2019-11-01T23:42:30.5889174Z 
2019-11-01T23:42:30.5994794Z ##[error]Bash exited with code '1'.
2019-11-01T23:42:30.6030786Z ##[section]Starting: Upload CPU usage statistics
2019-11-01T23:42:30.6033906Z ==============================================================================
2019-11-01T23:42:30.6033993Z Task         : Bash
2019-11-01T23:42:30.6034073Z Description  : Run a Bash script on macOS, Linux, or Windows
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

bors commented:
:broken_heart: Test failed - [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12510)
<!-- homu: {"type":"BuildFailed","builder_url":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12510","builder_name":"checks-azure"} -->

Mark-Simulacrum committed:
This disables the asmjs builder for beta

This builder is just a test builder, so the artifacts we produce
should be unaffected. Unfortunately, the mozilla-games S3 bucket appears
to have gone offline or at least is giving us forbidden errors, so the
asmjs builder isn't able to build anymore.

Mark-Simulacrum commented:
@bors r=pietroalbini

Just pushed up a commit that per discussion on Discord with @alexcrichton should be fine to land on beta -- it disables the testing for asmjs since the mozilla-games S3 bucket has gone offline. Master is still getting tested, but slightly differently, due to emscripten there being newer, which means we don't have this problem there.

bors commented:
:pushpin: Commit 710e468c0da3e7fbaeb05bf7ecf653c59537d3c8 has been approved by `pietroalbini`

<!-- @bors r=pietroalbini 710e468c0da3e7fbaeb05bf7ecf653c59537d3c8 -->
<!-- homu: {"type":"Approved","sha":"710e468c0da3e7fbaeb05bf7ecf653c59537d3c8","approver":"pietroalbini"} -->

bors commented:
:hourglass: Testing commit 710e468c0da3e7fbaeb05bf7ecf653c59537d3c8 with merge 32235c77259e2a814a90e0fc7ddecaa4bf0da36d...
<!-- homu: {"type":"BuildStarted","head_sha":"710e468c0da3e7fbaeb05bf7ecf653c59537d3c8","merge_sha":"32235c77259e2a814a90e0fc7ddecaa4bf0da36d"} -->

bors commented:
:boom: Test timed out
<!-- homu: {"type":"TimedOut"} -->

Mark-Simulacrum commented:
@bors retry

bors commented:
:hourglass: Testing commit 710e468c0da3e7fbaeb05bf7ecf653c59537d3c8 with merge 5242afe811c2648baf158eee28143c9facc350ee...
<!-- homu: {"type":"BuildStarted","head_sha":"710e468c0da3e7fbaeb05bf7ecf653c59537d3c8","merge_sha":"5242afe811c2648baf158eee28143c9facc350ee"} -->

bors commented:
:sunny: Test successful - [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12544)
Approved by: pietroalbini
Pushing 5242afe811c2648baf158eee28143c9facc350ee to beta...
<!-- homu: {"type":"BuildCompleted","approved_by":"pietroalbini","base_ref":"beta","builders":{"checks-azure":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12544"},"merge_sha":"5242afe811c2648baf158eee28143c9facc350ee"} -->
